### DISCUSSION TITLE
Upgrading decision

### DISCUSSION THREAD

**Discussion by hardysabs2** [visgl/deck.gl#7089] [Q&A]
I am trying to understand: 

a) The level of work involved in upgrading from 7.2.6 to 8.7.12, in particular currently being integrated with  react-map-gl (^5.0.10) and with currently implemented clustering using  an IconClusterLayer class that extends @deck.gl/core's CompositeLayer.  I know after upgrading the current clustering implementation simply does not work.  (I think I may have found some documentation for upgrade but if anything to add - greatly appreciated)

AND - perhaps more importantly...

b) What are the consequences and the timing of those consequences of NOT upgrading?   Will something for any reason suddenly cease to work out of the blue?  Bugs which only arise after other package changes or upgrades? Or simply cease to support etc...?

AND finally

c) What are the key advantages to upgrading?  (especially if in relation to clustering)  more features?  more efficiency?  fixes?  simpler implementation?
   
Many thanks indeed,

Andrew

**Comment by ibgreen** [comment]
I trust you have consulted the [upgrade guide](https://deck.gl/docs/upgrade-guide) which we put quite a bit of effort in to help guide such decisions..

You'll want to pay special attention to breaking changes, which we try to only do when we release major upgrades, see e.g.: https://deck.gl/docs/upgrade-guide#removed

Generally you are not going to get as much support on older releases. We typically fix confirmed bugs on latest release, but only rarely do we patch older branches.

We do not expect v7 to "stop working" that said browsers and JS tool chains (transpilers, linters, bundlers etc) are constantly evolving and the latest releases is generally where we make efforts to make things work with the latest tooling and optimize for new browser developments etc.

**Comment by hardysabs2** [comment]
@ibgreen Firstly - thank you so much for your response.  I appreciate it.

I did consult the upgrade guide thank you, though it was a little while ago now - I was moved on to something else.  I don't recall detail specifically about consequences of not upgrading, but I do need to revisit it TBF.

I shall revisit the guide and probably bite the bullet and upgrade.  I don't recall much in the guide at the time about clustering implementation.  In the current implementation I adapted a provided example.  If you think it's all in the guide for clustering then I shall search it out, but if you think there's more to know that would be useful to me or if there is an up to date example of clustering similar to the method I was using (react-map-gl (^5.0.10) and with currently implemented clustering using an IconClusterLayer class that extends @deck.gl/core's CompositeLayer) that I could adapt, then I'd really appreciate being pointed to those.

Many thanks again,

Andrew

**Reply by ibgreen** [reply]
react-map-gl has been extensively updated in v7 but you don't need to upgrade react-map-gl to upgrade deck.gl - that is a separate decision you will need to make based on info provided in that repo.

Otherwise why not just give it a try and see if your custom layer still works. I am not aware of any structural changes in those parts.

**Reply by hardysabs2** [reply]
Just to be clear I definitely already upgraded deckgl and it did fail, literally blank screen and lots of console failures if I recall correctly.  So I reverted.

From what you are saying, while there are breaking changes, you are not expecting any particular issues, at least not for the way clustering works, so if I simply follow the guide and do nothing else, I should not encounter any issues with clustering or anything else, and should not need any additional knowledge right?

How would you rate the time and effort in upgrading, roughly?

Many thanks again.

Andrew
